### ISSUE TITLE
model selection code for 7 models in paper?

### ISSUE THREAD

**Issue by fkgruber** [KarchinLab/bigmhc#11]
Hi
In the paper you selected 7 models optimized by learning rate and batch size. However, the train function provided here seems to just train a model for a fixed learning rate and batch size. 
Are there any code showing how the 7 models were created?

From the paper it seem that 7 batch sizes were used: 512  1024  2048  4096  8192 16384 32768. But how about the learning rate? Did you use a grid so that for each batch size you tested different learning rates and then you pick the epoch/learning rate that has the best performance on the validation? 

thanks
FKG

**Comment by benjaminalbert** [comment]
Hi @fkgruber, the learning rate was initially set to 1e-5, but we found that 5e-5 appeared better on the validation sets, so we manually identified and chose that for all batch sizes. Ideally, one would optimize this hyperparameter for each batch size individually, but we went with a constant learning for all models for the paper.

Then, we simply ran the predict script on the validation set using all the saved models to identify the best epoch per batch size. This was done in the terminal just using a bash for loop, and then we identified the best rather manually in a python terminal.
